packages/side-nav/src/SideNavItem.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { MenuItem, MenuItemProps } from '@leafygreen-ui/menu';
```

Comment:
So there's one main reason I think we might consider not using MenuItem as the base here: We lose a lot of control over the transition between states of the MenuItem.

I think with the design of these, we have a great micro-interaction opportunity in animating the scale of the background between different states, but (barring some hacky implementation), those animations will be difficult to add without full control of the rendered DOM.

Additionally, there's the added benefit of not needing to modify MenuItem for this component which is nice for two reasons:
1. We don't need to expose a className prop for MenuItem, which we want to avoid style overrides on in general.
2. There won't be conflicts with Brooke's work.

Open to thoughts on this, of course.